client/packages/programs/src/JsonForms/components/ProgramEvent.tsx

```diff
@@ -0,0 +1,153 @@
+import React from 'react';
+import { ControlProps, rankWith, uiTypeIs } from '@jsonforms/core';
+import { withJsonFormsControlProps } from '@jsonforms/react';
+import {
+  DetailInputWithLabelRow,
+  NumericTextInput,
+} from '@openmsupply-client/common';
+import {
+  DefaultFormRowSpacing,
+  DefaultFormRowSx,
+  FORM_LABEL_WIDTH,
+  useZodOptionsValidation,
+} from '../common';
+import {
+  useProgramEnrolments,
+  useProgramEvents,
+} from '../../api';
+
+import { z } from 'zod';
+
+export const programEventTester = rankWith(10, uiTypeIs('ProgramEvent'));
```

Comment:
Couldn't refactor EncounterProgramEvent to load this and caused endless rendering... so have moved it out to its own component...

Reply:
What was the problem, the EncounterProgramEvent should work here shouldn't it?

Reply:
The EncounterProgramEvent queries for the encounter by URL id then uses that to load the event. I tried using a ternary operator to load the ProgramEnrolment event but didn't seem to work and caused the api call for encounters to be called every second or so....

Reply:
[9a4951d](https://github.com/openmsupply/open-msupply/pull/1466/commits/9a4951d1af376a34a117d8b5bd7f926958c636c2)